js/memory.js

```diff
@@ -0,0 +1,76 @@
+// function createGame(){
+// 	new squareFactory()
+// }
+var squareFactory;
+(squareFactory = function squareFactory(){
+	var squareObjects = []
+	this.squareElements = $('.square');
+	this.squareObjects = squareObjects;
+	this.initSquareObjects = function initSquareObjects() {
+		for (var idx in this.squareElements) {
+			var square = this.squareElements[idx];
+
+			var object = new squareObject(square);
+			object.setClickEvent();
+			this.squareObjects.push(object)
+		};
+	}
+})()
```

Comment:
Hmm, you're both assigning the function to a variable and evaluating it all at once...I'd consider this a bit too "clever", and break those two things up so that it's more clear.